### ISSUE TITLE
Flickering in some games.

### ISSUE THREAD

**Issue by Zusion** [mgba-emu/mgba#76]
![](https://cloud.githubusercontent.com/assets/4545124/9090423/bc957710-3bcc-11e5-9899-87d1b0c6c8d2.gif)

Some games such as the one in the example above (Boktai) swap between 2 frames quickly to create a kind of pseudo transparency. I'm assuming this doesn't work as intended due to the difference in speed of the GBA's display compared to the monitors we use. Is it possible to implement some kind of interpolation or something to mimic that effect?

**Comment by endrift** [comment]
If you run with sync to video enabled, it'll look fine, but there are plenty of other issues there. As for deblinking, I'm not entirely sure what the best way to do that is without making everything look smudged. I can look into it for a future version.

**Comment by Zusion** [comment]
I've tried it with sync to video enabled and it's still the same.

**Comment by ShadowOne333** [comment]
I can confirm this issue for the Wii version.
I tried Yoshi's Island (Super Mario Advance 3) with the flickering question clouds and they don't flicker at all.

Either they're completely visible or completely invisible depending on when the flickering start.

**Comment by ShadowOne333** [comment]
Just a heads up.
I tried the nightly for Oct 24th and played Yoshi's Island with it.
The flickering clouds with the question mark now blink/flicker properly!

Seems to have been solved with that build.

**Comment by taisel** [comment]
@shadowone333 for 60 fps shadow flicker to not pulse at all would require the host display to be running at 30 fps and not 60 fps. Is the wii version running at 30 fps NTSC instead of 60?

**Comment by photodiode** [comment]
I encountered the flickering when I tried to play ZAS
Setting the frame skip to 1 "fixes" is, but then you only get half the background so to speak.

![zas2](https://cloud.githubusercontent.com/assets/16804570/21383939/8bbd7b9e-c7ab-11e6-8a87-bf74c52a777e.gif)

**Comment by sczther** [comment]
ZAS is intentionally, it's made for the original DMG, which has the last frame still drawn, so it essentially mixes them both. (That's the famous ghosting, making many action games unpleasant to play.) ZAS uses this quirk to have rich backgrounds, on a DMG, it has both the ships and the background at once. There should be an option like "mix frames" in mGBA to simulate this behavior.  If there isn't, then that's the problem.

**Comment by photodiode** [comment]
Yes, a "mix the current and last frame" option would be great.

**Comment by LordGinger** [comment]
The issue is still present in the latest nightly (0.6-4464-a5f029c).
In other emulators it can be alleviated using a motion blur shader, which helps with interframe blending.

**Comment by endrift** [comment]
"Issue"? It's an intended effect on hardware. The only reason this is still open is so I can remember to add an interframe blending option to imitate hardware ghosting.

**Comment by GrieverV** [comment]
Just noting that Final Fantasy I & II: Dawn of Souls also uses this effect for transparency
![Final Fantasy I](https://user-images.githubusercontent.com/38054248/43141680-824a4bb8-8f24-11e8-8d58-058baeb40c10.gif)

**Comment by ubergeek77** [comment]
I know this is an old issue, but I thought I would add something that @endrift and others may find interesting.

When I switched from playing Boktai on my Android phone to playing it on Windows (both using the mGBA core), the flickering became much more noticeable on Windows. I double checked on my phone, and sure enough, the transparency/flickering effect is quite good under Android. I didn't even give the effect much thought.

I don't know any of the specifics, but I remember learning a long time ago that Windows has its own sort of Vsync/framebuffer that it applies to everything running in windowed mode. From personal experience, this effect has been noticeable; I remember having a ton of screen tearing in older games without proper Vsync support, and running them in windowed/borderless windowed solved most of the issues, whereas exclusive fullscreen would bypass this framebuffer, leading to screen tearing. I tried running mGBA in fullscreen, but either it only implements borderless windowed mode, or my having multiple monitors prevented me from trying it in exclusive fullscreen mode. Doing so had no visible affect.

In any case, I think this Windows framebuffer is causing the problems here. Windows erroneously trying to sync the frames to the screen's refresh rate are causing some of the flickering effects in GBA games to be skipped. Using the mGBA setting to sync to video does help somewhat, but the improvement is minimal, and it's nowhere near as seamless as on Android. Again, I don't know the full technical details of this, but from what I've seen, I'd say this is a pretty good guess.

In summary:
- Using **Windows** to run mGBA on RetroArch **yields issues** with transparency/flickering
- Using **Windows** to run standalone mGBA **yields issues** with transparency/flickering
- Using **Android** to run mGBA on RetroArch ***does not*** **yield issues** with transparency/flickering (they work perfectly)

I'm not sure about other consoles that can run mGBA through RetroArch, but if I had to guess, this is a Windows-specific issue.

**Comment by coolanybody** [comment]
This also happens with Friends of Mineral Town
![bad text](https://user-images.githubusercontent.com/34614648/56446288-e28aa480-62cf-11e9-9427-8c0b4e0dcb41.gif)

**Comment by endrift** [comment]
Interframe blending for Qt and SDL versions added in 47bf26ff73d9bdf8a044773b20fc816f6e823e7e. Wii, Vita, 3DS and Switch implementations forthcoming. Still deciding how I'm going to handle RA.

**Comment by CharlesCorrin** [comment]
A wonderful feature. I tested in Golden Sun and Final Fantasy I & II - Dawn of Souls. Works great.

Except... I think I found a bug. Turn on the interframe blending. Then run your mouse cursor over the screen. It seems to interfere with the blending when you do. This is in software and OpenGL video modes both.

**Comment by endrift** [comment]
Implemented now everywhere except RA and OpenEmu.

**Comment by benderscruffy** [comment]
everytime i tick Interframe blending the screen goes black but music plays (every game)
i have an AMD Radeon HD 7450

**Comment by CharlesCorrin** [comment]
If it helps, my video card is the Nvidia Geforce 750 Ti and I'm working with Windows 10 Home 64-bit Version 1809.

**Comment by endrift** [comment]
@CharlesCorrin what version of the drivers do you have? I can't reproduce on my (newer) Nvidia card on Windows.

**Comment by CharlesCorrin** [comment]
Nvidia driver version 26.21.14.3086 , dated May 22 2019. The newest one at this time. I usually get the newest drivers fast. Could the driver be buggy?

**Comment by endrift** [comment]
I encountered driver bugs with an earlier version, but I doubt that's what's going on here. I might need to find a way to log this stuff.

**Comment by CharlesCorrin** [comment]
After the success with https://github.com/mgba-emu/mgba/commit/09ccf5403eb0a0d85fa119c612b9070fe8807818 for the other issue, I decided I should test this. The bug I reported in this topic is fixed too. :3 Though I don't know why it is.

benderscruffy, how's the one you reported? Fixed or no?

**Comment by benderscruffy** [comment]
no it is worse now
any game using opengl is black screen with music playing  with or without Interframe blending ticked
i can only use software (Qt) now

**Comment by endrift** [comment]
OpenGL force 1.x should be fixed now, as is interframe blending on 1.x.

**Comment by endrift** [comment]
This is fixed now in everything except the libretro core, but it's a frontend feature so Retroarch should implement it instead of the cores.

**Comment by tnelling** [comment]
![mgbaAW2](https://user-images.githubusercontent.com/15524479/84745800-1704f980-af83-11ea-860f-64e1f94c4c5d.gif)

Not sure if this is the exact same issue, but I ran into this with Advance Wars 2 dialogue. The issue persists whether or not I have interframe blending or sync to video enabled (in the gif both are enabled).

**Comment by endrift** [comment]
That is issue #1790, not this one. Do you have an AMD GPU?

**Comment by tnelling** [comment]
I'm not sure why it would be #1790, since I'm not holding the turbo key (or doing anything turbo related).

I have an NVIDIA GeForce.

**Comment by Liger0** [comment]
> 
> 
> I'm not sure why it would be #1790, since I'm not holding the turbo key (or doing anything turbo related).
> 
> I have an NVIDIA GeForce.

I actually have had it without using turbo hold key, but it seems to be more reproduce-able with the turbo.

**Comment by tnelling** [comment]
For what it's worth, after I created this, I noticed that it was only happening with dialogue at the top of the screen. The dialogue at the bottom didn't seem to have the issue. Though I can't say with absolute certainty that it never did.

**Comment by cjlaborde** [comment]
> ![mgbaAW2](https://user-images.githubusercontent.com/15524479/84745800-1704f980-af83-11ea-860f-64e1f94c4c5d.gif)
> 
> Not sure if this is the exact same issue, but I ran into this with Advance Wars 2 dialogue. The issue persists whether or not I have interframe blending or sync to video enabled (in the gif both are enabled).

I have same issue with advanced wars 
![flickering3](https://user-images.githubusercontent.com/40572114/88259933-56fd8180-cc91-11ea-9c3a-bf7b2cb96945.gif)

**Comment by endrift** [comment]
Again, that's #1790, not this.

**Comment by endrift** [comment]
Actually, looking at these GIFs again, that might be a separate bug that just affects Advance Wars. It's 100% _not_ this bug, but it may not be that bug either.

**Comment by endrift** [comment]
Relevant Advance Wars bug was filed at #1823.
